dot/network/warp_sync.go

```diff
@@ -16,12 +16,12 @@ import (
 type WarpSyncProvider interface {
 	// Generate proof starting at given block hash. The proof is accumulated until maximum proof
 	// size is reached.
-	generate(start common.Hash) (encodedProof []byte, err error)
+	Generate(start common.Hash) (encodedProof []byte, err error)
 }
 
 func (s *Service) handleWarpSyncRequest(req messages.WarpProofRequest) ([]byte, error) {
```

Comment:
I would suggest to add a spamming rule here and decrease peers that spam warp proof requests, you can hash the message and cache it to check if we are receiving the same message from the same peer and define a limit for it.

Reply:
Feel free to open an issue for this and address in another PR.